rts/motoko-rts/src/gc/copying.rs

```diff
@@ -79,6 +81,16 @@ pub unsafe fn copying_gc_internal<
         );
     }
 
+    if (*region0_ptr_loc).is_ptr() {
```

Comment:
I could later, independent of this PR, refactor the root set collection, to avoid duplicate logic for handling the additional "region 0" root in all of the GCs.